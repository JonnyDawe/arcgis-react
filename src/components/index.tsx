export { ArcLayer, useLayerView } from './ArcLayer/ArcLayer';
export { ArcUI } from './ArcUI/ArcUI';
export { ArcMapView, ArcSceneView } from './ArcView/ArcView';
export { useMapView, useSceneView } from './ArcView/ViewContext';
export { ArcWidget } from './ArcWidget/ArcWidget';
