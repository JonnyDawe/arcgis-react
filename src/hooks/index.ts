export { useWatchEffect, useOnEvent } from './useWatchEffect';
export { useWidget } from './useWidget';
