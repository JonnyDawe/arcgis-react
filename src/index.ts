export { MapContext } from './components/MapViewComponent';
export { MapStateProvider } from './hooks/MapStateContext';
