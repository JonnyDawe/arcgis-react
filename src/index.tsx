export * from './components/ArcLayer/generated/ArcBaseDynamicLayer';
export * from './components/ArcLayer/generated/ArcBaseElevationLayer';
export * from './components/ArcLayer/generated/ArcBaseTileLayer';
export * from './components/ArcLayer/generated/ArcBingMapsLayer';
export * from './components/ArcLayer/generated/ArcBuildingSceneLayer';
export * from './components/ArcLayer/generated/ArcCSVLayer';
export * from './components/ArcLayer/generated/ArcDimensionLayer';
export * from './components/ArcLayer/generated/ArcElevationLayer';
export * from './components/ArcLayer/generated/ArcFeatureLayer';
export * from './components/ArcLayer/generated/ArcGeoJSONLayer';
export * from './components/ArcLayer/generated/ArcGeoRSSLayer';
export * from './components/ArcLayer/generated/ArcGraphicsLayer';
export * from './components/ArcLayer/generated/ArcGroupLayer';
export * from './components/ArcLayer/generated/ArcImageryLayer';
export * from './components/ArcLayer/generated/ArcImageryTileLayer';
export * from './components/ArcLayer/generated/ArcIntegratedMeshLayer';
export * from './components/ArcLayer/generated/ArcKMLLayer';
export * from './components/ArcLayer/generated/ArcKnowledgeGraphLayer';
export * from './components/ArcLayer/generated/ArcLayer';
export * from './components/ArcLayer/generated/ArcLineOfSightLayer';
export * from './components/ArcLayer/generated/ArcMapImageLayer';
export * from './components/ArcLayer/generated/ArcMapNotesLayer';
export * from './components/ArcLayer/generated/ArcMediaLayer';
export * from './components/ArcLayer/generated/ArcOGCFeatureLayer';
export * from './components/ArcLayer/generated/ArcOpenStreetMapLayer';
export * from './components/ArcLayer/generated/ArcPointCloudLayer';
export * from './components/ArcLayer/generated/ArcRouteLayer';
export * from './components/ArcLayer/generated/ArcSceneLayer';
export * from './components/ArcLayer/generated/ArcStreamLayer';
export * from './components/ArcLayer/generated/ArcSubtypeGroupLayer';
export * from './components/ArcLayer/generated/ArcTileLayer';
export * from './components/ArcLayer/generated/ArcUnknownLayer';
export * from './components/ArcLayer/generated/ArcUnsupportedLayer';
export * from './components/ArcLayer/generated/ArcVectorTileLayer';
export * from './components/ArcLayer/generated/ArcVoxelLayer';
export * from './components/ArcLayer/generated/ArcWCSLayer';
export * from './components/ArcLayer/generated/ArcWebTileLayer';
export * from './components/ArcLayer/generated/ArcWFSLayer';
export * from './components/ArcLayer/generated/ArcWMSLayer';
export * from './components/ArcLayer/generated/ArcWMTSLayer';
export { ArcUI } from './components/ArcUI/ArcUI';
export { ArcMapView } from './components/ArcView/ArcMapView';
export { ArcSceneView } from './components/ArcView/ArcSceneView';
export { useMapView, useSceneView } from './components/ArcView/ViewContext';
export * from './components/ArcWidget/generated/ArcAreaMeasurement2D';
export * from './components/ArcWidget/generated/ArcAreaMeasurement3D';
export * from './components/ArcWidget/generated/ArcAttachments';
export * from './components/ArcWidget/generated/ArcAttribution';
export * from './components/ArcWidget/generated/ArcBasemapGallery';
export * from './components/ArcWidget/generated/ArcBasemapLayerList';
export * from './components/ArcWidget/generated/ArcBasemapToggle';
export * from './components/ArcWidget/generated/ArcBookmarks';
export * from './components/ArcWidget/generated/ArcBuildingExplorer';
export * from './components/ArcWidget/generated/ArcCompass';
export * from './components/ArcWidget/generated/ArcCoordinateConversion';
export * from './components/ArcWidget/generated/ArcDaylight';
export * from './components/ArcWidget/generated/ArcDirections';
export * from './components/ArcWidget/generated/ArcDirectLineMeasurement3D';
export * from './components/ArcWidget/generated/ArcDistanceMeasurement2D';
export * from './components/ArcWidget/generated/ArcEditor';
export * from './components/ArcWidget/generated/ArcElevationProfile';
export * from './components/ArcWidget/generated/ArcExpand';
export * from './components/ArcWidget/generated/ArcFeature';
export * from './components/ArcWidget/generated/ArcFeatureForm';
export * from './components/ArcWidget/generated/ArcFeatureTable';
export * from './components/ArcWidget/generated/ArcFeatureTemplates';
export * from './components/ArcWidget/generated/ArcFloorFilter';
export * from './components/ArcWidget/generated/ArcFullscreen';
export * from './components/ArcWidget/generated/ArcHistogram';
export * from './components/ArcWidget/generated/ArcHistogramRangeSlider';
export * from './components/ArcWidget/generated/ArcHome';
export * from './components/ArcWidget/generated/ArcLayerList';
export * from './components/ArcWidget/generated/ArcLegend';
export * from './components/ArcWidget/generated/ArcLineOfSight';
export * from './components/ArcWidget/generated/ArcLocate';
export * from './components/ArcWidget/generated/ArcMeasurement';
export * from './components/ArcWidget/generated/ArcNavigationToggle';
export * from './components/ArcWidget/generated/ArcPopup';
export * from './components/ArcWidget/generated/ArcPrint';
export * from './components/ArcWidget/generated/ArcScaleBar';
export * from './components/ArcWidget/generated/ArcScaleRangeSlider';
export * from './components/ArcWidget/generated/ArcSearch';
export * from './components/ArcWidget/generated/ArcShadowCast';
export * from './components/ArcWidget/generated/ArcSketch';
export * from './components/ArcWidget/generated/ArcSlice';
export * from './components/ArcWidget/generated/ArcSlider';
export * from './components/ArcWidget/generated/ArcSwipe';
export * from './components/ArcWidget/generated/ArcTableList';
export * from './components/ArcWidget/generated/ArcTimeSlider';
export * from './components/ArcWidget/generated/ArcTrack';
export * from './components/ArcWidget/generated/ArcUtilityNetworkAssociations';
export * from './components/ArcWidget/generated/ArcUtilityNetworkTrace';
export * from './components/ArcWidget/generated/ArcWeather';
export * from './components/ArcWidget/generated/ArcWidget';
export * from './components/ArcWidget/generated/ArcZoom';
export {
  useArcState,
  useOnEvent,
  useWatchEffect,
  useWatchState,
  useWhenEffect,
} from './hooks/useWatchEffect';
