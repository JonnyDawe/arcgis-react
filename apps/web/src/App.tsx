import arcgisDarkCss from '@arcgis/core/assets/esri/themes/dark/main.css?inline';
import arcgisLightCss from '@arcgis/core/assets/esri/themes/light/main.css?inline';
import {
  CalciteAction,
  CalciteActionBar,
  CalciteLoader,
  CalciteNavigation,
  CalciteNavigationLogo,
  CalciteShell,
  CalciteShellPanel,
} from '@esri/calcite-components-react';
import { lazy, Suspense, useEffect, useState } from 'react';

import logoDark from './assets/arcgis-react-logo-dark.png';
import logoLight from './assets/arcgis-react-logo-light.png';
import GithubIcon from './assets/GithubIcon';
import { CodeDisplay, CodeDisplayAsync } from './components/CodeDisplay';
import { useTheme } from './contexts/ThemeProvider';
import { ActionItem, useCalciteActionBar } from './hooks/calciteHooks';

const Examples: ActionItem[] = [
  {
    name: 'Home',
    component: lazy(() => import('./examples/Home')),
    icon: 'home',
  },
  {
    name: 'Map View',
    component: lazy(() => import('./examples/MapView')),
    code: () => import('./examples/MapView?raw'),
    icon: 'map',
  },
  {
    name: 'Scene View',
    component: lazy(() => import('./examples/SceneView')),
    code: () => import('./examples/SceneView?raw'),
    icon: 'globe',
  },
  {
    name: 'ReactiveUtils',
    component: lazy(() => import('./examples/ReactiveUtils')),
    code: () => import('./examples/ReactiveUtils?raw'),
    icon: 'refresh',
  },
  {
    name: 'Daylight Widget',
    component: lazy(() => import('./examples/DaylightWidget')),
    code: () => import('./examples/DaylightWidget?raw'),
    icon: 'brightness',
  },
  {
    name: 'Intro to Layers',
    component: lazy(() => import('./examples/IntroToLayers')),
    code: () => import('./examples/IntroToLayers?raw'),
    icon: 'layers',
  },
  {
    name: 'Sketch Widget',
    component: lazy(() => import('./examples/SketchWidget')),
    code: () => import('./examples/SketchWidget?raw'),
    icon: 'pencil',
  },
  {
    name: 'Client-Side Stream Layer',
    component: lazy(() => import('./examples/ClientSideStreamLayer')),
    code: () => import('./examples/ClientSideStreamLayer?raw'),
    icon: 'layer-graphics',
  },
  {
    name: 'Basemap Picker',
    component: lazy(() => import('./examples/BasemapPicker')),
    code: () => import('./examples/BasemapPicker?raw'),
    icon: 'basemap',
  },
  {
    name: 'MultiView',
    component: lazy(() => import('./examples/MultiView')),
    code: () => import('./examples/MultiView?raw'),
    icon: '3d-glasses',
  },
  {
    name: 'Custom Zoom Component',
    component: lazy(() => import('./examples/CustomZoom')),
    code: () => import('./examples/CustomZoom?raw'),
    icon: 'magnifying-glass',
  },
  // {
  //   name: 'FeatureSelection',
  //   component: lazy(() => import('./examples/FeatureSelection')),
  //   code: () => import('./examples/FeatureSelection?raw'),
  //   icon: 'select',
  // },
];

export function App() {
  const { theme, setTheme } = useTheme();

  // Set the ArcGIS theme on the document head
  useEffect(() => {
    const style = document.createElement('style');
    style.innerHTML = theme === 'dark' ? arcgisDarkCss : arcgisLightCss;
    document.head.appendChild(style);

    return () => {
      document.head.removeChild(style);
    };
  }, [theme]);

  const { currentAction, actions } = useCalciteActionBar(
    Examples,
    window.location.hash
      ? decodeURI(window.location.hash.slice(1))
      : Examples[0].name
  );

  useEffect(() => {
    if (currentAction) window.location.hash = currentAction?.name;
  }, [currentAction]);

  return (
    <div style={{ colorScheme: theme }}>
      <CalciteShell className={`calcite-mode-${theme}`}>
        <CalciteNavigation slot="header">
          <CalciteNavigationLogo
            slot="logo"
            heading="ArcGIS React"
            thumbnail={theme === 'light' ? logoLight : logoDark}
            style={{
              '--calcite-font-size-0': '20px',
            }}
          />
          <div slot="content-end">
            <CalciteAction
              text="View on Github"
              onClick={() =>
                window.open(
                  'https://github.com/am-maneaters/arcgis-react',
                  '_blank'
                )
              }
              scale="l"
            >
              <GithubIcon />
            </CalciteAction>
            <CalciteAction
              icon={theme === 'dark' ? 'brightness' : 'moon'}
              text="Toggle theme"
              onClick={() => setTheme(theme === 'dark' ? 'light' : 'dark')}
              scale="l"
            />
          </div>
        </CalciteNavigation>

        <CalciteShellPanel slot="panel-start" collapsed>
          <CalciteActionBar slot="action-bar" expanded>
            {actions}
          </CalciteActionBar>
        </CalciteShellPanel>

        {currentAction && (
          <Suspense
            fallback={
              <div className="w-full h-full">
                <CalciteLoader label="Sample Loading" />
              </div>
            }
          >
            <div className="flex flex-col gap-4 p-4 bg-dotted min-h-full items-center box-border [&>*]:max-w-3xl [&>*]:w-full">
              <div className="flex-1 min-h-[50vh] rounded-lg overflow-hidden shadow-3xl bg-foreground-1">
                {currentAction?.component && <currentAction.component />}
              </div>
              <div className="shadow-3xl overflow-auto rounded-lg">
                {currentAction.code && (
                  <CodeDisplayAsync codePromise={currentAction?.code} />
                )}
              </div>
            </div>
          </Suspense>
        )}
      </CalciteShell>
    </div>
  );
}
