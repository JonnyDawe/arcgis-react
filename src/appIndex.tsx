import React, { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import '@arcgis/core/assets/esri/themes/dark/main.css';

import { MapStateProvider } from './hooks/MapStateContext';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { TestsRouter } from './pages/ReactiveDemo';

// Create a root element for the application
const root = createRoot(document.querySelector('#root')!);

const queryClient = new QueryClient({
  defaultOptions: { queries: { refetchOnWindowFocus: false } },
});
// Render the application
root.render(
  <StrictMode>
    <MapStateProvider>
      <QueryClientProvider client={queryClient}>
        <TestsRouter />
      </QueryClientProvider>
    </MapStateProvider>
  </StrictMode>
);
