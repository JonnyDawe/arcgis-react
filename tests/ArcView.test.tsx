import { cleanup, render, screen } from '@testing-library/react';
import { describe, expect, it, vitest } from 'vitest';

import { ArcMapView, ArcSceneView } from '../src';

// lol the arcgis api doesn't seem to have any good testing resources...
vitest.mock('@arcgis/core/views/MapView', async () => ({
  default: class MapView {
    constructor() {}

    when() {
      return Promise.resolve();
    }
  },
  // your mocked methods
}));
vitest.mock('@arcgis/core/views/SceneView', async () => ({
  default: class SceneView {
    constructor() {}
    when() {
      return Promise.resolve();
    }
  },

  // your mocked methods
}));
vitest.mock('@arcgis/core/WebMap', () => ({
  default: class WebMap {
    constructor() {
      return {};
    }
  },
}));

describe('ArcView', () => {
  it('renders sceneview', () => {
    const { container } = render(<ArcSceneView />);
    expect(container).toMatchSnapshot();
    cleanup();
    expect(container).toMatchSnapshot();
  });

  it('renders mapview', () => {
    const { container } = render(<ArcMapView />);
    expect(container).toMatchSnapshot();
    cleanup();
    expect(container).toMatchSnapshot();
  });
});
