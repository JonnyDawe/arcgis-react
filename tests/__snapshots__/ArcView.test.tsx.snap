// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders mapview 1`] = `
<div>
  <div />
</div>
`;

exports[`App > renders mapview 2`] = `<div />`;

exports[`App > renders sceneview 1`] = `
<div>
  <div />
</div>
`;

exports[`App > renders sceneview 2`] = `<div />`;

exports[`ArcView > renders mapview 1`] = `
<div>
  <div />
</div>
`;

exports[`ArcView > renders mapview 2`] = `<div />`;

exports[`ArcView > renders sceneview 1`] = `
<div>
  <div />
</div>
`;

exports[`ArcView > renders sceneview 2`] = `<div />`;
